Start some docs for the operator (#2076)
kroxylicious/kroxylicious: docs/modules/configuring/con-configuring-kafkaservice-protocol.adoc

[id='con-configuring-kafkaservice-protocol-{context}']

> = Controlling the TLS version used

Reviewer: ```suggestion
= Controlling TLS versions
```

> Older versions of TLS (and SSL before it) are now considered insecure.

Reviewer: ```suggestion
Some older versions of TLS (and SSL before it) are now considered insecure.
```

> IMPORTANT: It is considered good practice to disable insecure versions of the protocol. You can only do this is the proxied Kafka cluster you want to connect to supports the version that you want to use.

Reviewer: ```suggestion
IMPORTANT: It is good practice to disable insecure protocol versions. Only do this if the proxied Kafka cluster supports the version that you want to use.
```

> .Example `KafkaService` configured so that the proxy will negotiate only a TLS v1.3 connection to `kafka.example.com`.

Reviewer: ```suggestion
This example configures a `KafkaService` to allow only TLS v1.3 and v1.2 when connecting to `kafka.example.com`.

.Example `KafkaService` with restricted TLS protocol versions
```

[source,yaml]
----
kind: KafkaService
metadata:
  # ...
spec:
  bootstrapServers: kafka.example.com:9092
  tls:
    # ...
    protocols: # <1>
      allowed: # <2>
        - TLSv1.3
        - TLSv1.2
----

> <1> The `protocols` object configures the TLS procotol versions

Reviewer: ```suggestion
<1> The `protocols` property configures the TLS protocol versions
```

<2> `allowed` lists the versions of TLS which are permitted.

> The `protocols` object also supports `denied` if you prefer to exclude the unpermitted versions, rather than including the permitted versions.

Reviewer: ```suggestion
The `protocols` property also supports `denied`, if you prefer to list the versions to exclude instead.
```
